packages/next/src/lib/metadata/resolve-metadata.ts

```diff
@@ -484,9 +486,16 @@ export async function resolveMetadataItems({
       : // Pass through parent params to children
         parentParams
 
-  const layerProps = {
-    params: currentParams,
-    ...(isPage && { searchParams }),
+  let layerProps
```

Comment:
Maybe it's valuable to add a type here to ensure it's not reassigned to something else?